content/300-guides/100-performance-and-optimization/090-metrics.mdx

```diff
@@ -0,0 +1,416 @@
+
+# HELP query_total_queries The total number of queries executed
+
+# TYPE query_total_queries counter
+
+query_total_queries 0
+
+# HELP pool_active_connections The number of active connections in use
+
+# TYPE pool_active_connections gauge
+
+pool_active_connections 0
+
+# HELP pool_idle_connections The number of connections that are not being used
+
+# TYPE pool_idle_connections gauge
+
+pool_idle_connections 21
+
+# HELP pool_wait_count The number of workers waiting for a connection
+
+# TYPE pool_wait_count gauge
+
+pool_wait_count
+
+# HELP query_active_transactions The number of active transactions
+
+# TYPE query_active_transactions gauge
+
+query_active_transactions
+
+# HELP pool_wait_duration_ms The wait time for a worker to get a connection.
+
+# TYPE pool_wait_duration_ms histogram
+
+pool_wait_duration_ms_bucket{le="0"} 0
+pool_wait_duration_ms_bucket{le="1"} 1
+pool_wait_duration_ms_bucket{le="5"} 1
+pool_wait_duration_ms_bucket{le="10"} 1
+pool_wait_duration_ms_bucket{le="50"} 1
+pool_wait_duration_ms_bucket{le="100"} 1
+pool_wait_duration_ms_bucket{le="500"} 1
+pool_wait_duration_ms_bucket{le="1000"} 1
+pool_wait_duration_ms_bucket{le="5000"} 1
+pool_wait_duration_ms_bucket{le="50000"} 1
+pool_wait_duration_ms_bucket{le="+Inf"} 1
+pool_wait_duration_ms_sum 0.050875000000000004
+pool_wait_duration_ms_count 1
+```
+
+### Use Prisma metrics with the Prometheus metrics system
+
+In the majority of cases, Prometheus must scrape an endpoint to retrieve metrics. The following example shows how to send data with `Express.js`:
+
+```js
+import { PrismaClient } from "@prisma/client";
+import express, { Request, Response } from "express";
+
+const app = express();
+const port = 4000;
+const prisma = new PrismaClient();
+
+app.get("/metrics", async (\_req, res: Response) => {
+let metrics = await prisma.$metrics.prometheus();
+res.end(metrics);
+});
+
+app.listen(port, () => {
+console.log(`Example app listening on port ${port}`);
+})
+```
+
+The following example shows how to combine Prisma metrics with other Prometheus client libraries:
+
+```js
+import { PrismaClient } from "@prisma/client";
+import express, { Request, Response } from "express";
+import prom from "prom-client";
+
+let registry = new prom.Registry();
+const app = express();
+const port = 4000;
+const prisma = new PrismaClient();
+
+app.get("/metrics", async (\_req, res: Response) => {
+let prismaMetrics = await prisma.$metrics.prometheus();
+let appMetrics = await registry.metrics();
+res.end(prismaMetrics + appMetrics);
+});
+
+app.listen(port, () => {
+console.log(`Example app listening on port ${port}`);
+});
+```
+
+## Global labels
+
+You can add global labels to your metrics to help you group and separate your metrics. Each instance of Prisma Client will add these labels to all the metrics that it generates. For example, you can group your metrics by infrastructure region, or by server, with a label like `{server: "us-server1", "app_version": "1"}`.
```

Comment:
Formatting of the label JSON is a bit weird. Is that necessary, or could it also be this:
```suggestion
You can add global labels to your metrics to help you group and separate your metrics. Each instance of Prisma Client will add these labels to all the metrics that it generates. For example, you can group your metrics by infrastructure region, or by server, with a label like `{ server: "us-server1", "app_version": "1" }`.
```
?